eclipse-collections-api/src/main/java/org/eclipse/collections/api/InternalIterable.java

```diff
@@ -49,6 +51,12 @@
     @SuppressWarnings("UnnecessaryFullyQualifiedName")
     void forEach(Procedure<? super T> procedure);
 
+    @Override
+    default void forEach(Consumer<? super T> action)
+    {
+        forEach((Procedure<? super T>) (x -> action.accept(x)));
```

Comment:
This is a good change.  I think it would be slightly better if you break the type out onto a separate line and use a method reference as follows.  

```
Procedure<? super T> procedure = consumer::accept;
this.forEach(procedure);
```

I would also recommend duplicating the functionality in the the forEach test in `AbstractRichIterableTestCase` and call it forEachConsumer so we know this code path is explicitly covered.